src/components/show-products/index.js

```diff
@@ -43,28 +51,56 @@ export const ShowProducts = () => {
 
   return (
     <ContentContainer>
-      <ul>
-        {products?.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
-          <li className="checkbox-item" key={id}>
-            <label htmlFor={id} className="checkbox-label">
-              <input
-                type="checkbox"
-                id={id}
-                name={productName}
-                onChange={() => handleOnChange(id)}
-                checked={
-                  lastPurchaseDate && compareDates(lastPurchaseDate.toDate())
-                }
-                ariaLabel={timeFrames[timeFrame]}
-              />
-              <span
-                className={`checkmark checkbox-timeFrame-${timeFrame}`}
-              ></span>
-              <span className="checkbox-name">{productName}</span>
-            </label>
-          </li>
-        ))}
-      </ul>
+      <form className="filter-form">
+        <label htmlFor="filter">
+          Filter items
+          <input
+            className="search-item"
+            type="search"
+            id="filter"
+            name="filter"
+            aria-label="Search through list content."
+            placeholder="Start typing here..."
+            value={searchTerm}
+            onChange={(event) => setSearchTerm(event.target.value)}
+          />
+        </label>
+        {searchTerm && (
+          <input
+            class="close-icon"
+            onClick={deleteSerachTerm}
+            type="reset"
+            value="X"
+          />
+        )}
```

Comment:
This is great! thanks for adding it :) 

Just one minor suggestion: It'd be great if you add an `aria-label` attribute here to describe the intention of this button. This will help screen readers to announce that this button will clear the contents of the search form.